diff --git a/lib/voice/VoiceConnection.js b/lib/voice/VoiceConnection.js
index c8f696a2d62ade810cc36edeef45a3dc6a49d96f..6f0773a49702f4ec1694a08bcc3320b6b3c47ae8 100644
--- a/lib/voice/VoiceConnection.js
+++ b/lib/voice/VoiceConnection.js
@@ -392,7 +392,7 @@ class VoiceConnection extends EventEmitter {
                             this.disconnect(err);
                         }
                     });
-                    const udpMessage = Buffer.allocUnsafe(74);
+                    const udpMessage = Buffer.alloc(74);
                     udpMessage.writeUInt16BE(0x1, 0);
                     udpMessage.writeUInt16BE(70, 2);
                     udpMessage.writeUInt32BE(this.ssrc, 4);
@@ -546,7 +546,12 @@ class VoiceConnection extends EventEmitter {
             if(this.connecting || this.ready) {
                 let reconnecting = true;
                 if(code === 4006) {
-                    reconnecting = false;
+                    if (this.channelID) {
+                        reconnecting = true;
+                        err = null;
+                    } else {
+                        reconnecting = false;
+                    }
                 } else if(code === 4014) {
                     if(this.channelID) {
                         data.endpoint = null;
@@ -914,10 +919,8 @@ class VoiceConnection extends EventEmitter {
             let dropPacket = false;
             if(this.daveSession && !data.equals(SILENCE_FRAME)) {
                 if(this.daveSession.ready && !userID) {
-                    this.emit("warn", "Packet was dropped due having an undefined user ID during a DAVE session");
                     dropPacket = true;
                 } else if(!this.daveSession.ready && this.daveProtocolVersion !== 0) {
-                    this.emit("warn", "Packet was dropped due to DAVE session not being ready yet");
                     dropPacket = true;
                 } else if(userID) {
                     try {
