diff --git a/lib/voice/VoiceConnection.js b/lib/voice/VoiceConnection.js
index 8078daae083ddd55f24b36f5881763c9b58a0549..f115b7834171c20e3fe4dc62af6e58e725822102 100644
--- a/lib/voice/VoiceConnection.js
+++ b/lib/voice/VoiceConnection.js
@@ -392,7 +392,7 @@ class VoiceConnection extends EventEmitter {
                             this.disconnect(err);
                         }
                     });
-                    const udpMessage = Buffer.allocUnsafe(74);
+                    const udpMessage = Buffer.alloc(74);
                     udpMessage.writeUInt16BE(0x1, 0);
                     udpMessage.writeUInt16BE(70, 2);
                     udpMessage.writeUInt32BE(this.ssrc, 4);
@@ -546,7 +546,12 @@ class VoiceConnection extends EventEmitter {
             if(this.connecting || this.ready) {
                 let reconnecting = true;
                 if(code === 4006) {
-                    reconnecting = false;
+                    if (this.channelID) {
+                        reconnecting = true;
+                        err = null;
+                    } else {
+                        reconnecting = false;
+                    }
                 } else if(code === 4014) {
                     if(this.channelID) {
                         data.endpoint = null;
@@ -911,6 +916,8 @@ class VoiceConnection extends EventEmitter {
             }
 
             const userID = this.ssrcUserMap[msg.readUIntBE(8, 4)];
+            if (!userID) return;
+            if (this.daveSession && !this.#daveReady) return;
             if(this.daveSession && userID && !data.equals(SILENCE_FRAME)) {
                 try {
                     const canDecrypt = this.#daveReady || (this.daveSession.ready && this.daveSession.canPassthrough(userID));
